src/main/java/com/minecolonies/rankup/util/AccountingUtils.java

```diff
@@ -0,0 +1,524 @@
+    {
+        final Connection conn = getConn();
+
+        if (conn != null)
+        {
+            Statement stmt = conn.createStatement();
+
+            if (isDate)
+            {
+                stmt.execute("UPDATE " + table_id + " SET " + column + " = " + toDate(attribute) + " WHERE " + UUID_COLUMN + " = '" + uuid + "'");
+            }
+            else
+            {
+                stmt.execute("UPDATE " + table_id + " SET " + column + " = '" + attribute + "' WHERE " + UUID_COLUMN + " = '" + uuid + "'");
+            }
+        }
+    }
+
+    private void updatePlayerAttribute(final UUID uuid, final String column, final int attribute) throws SQLException
+    {
+        final Connection conn = getConn();
+
+        if (conn != null)
+        {
+            Statement stmt = conn.createStatement();
+
+            stmt.execute("UPDATE " + table_id + " SET " + column + " = " + attribute + " WHERE " + UUID_COLUMN + " = '" + uuid + "'");
+        }
+    }
+
+    public void updatePlayerName(final UUID uuid, final String name)
+    {
+        try
+        {
+            if (plugin.getModuleContainer().isModuleLoaded(DatabaseModule.ID))
+            {
+                updatePlayerAttribute(uuid, PLAYER_NAME_COLUMN, name, false);
+            }
+            else
+            {
+                final AccountConfigData accConfig = getAccountConfig();
+                accConfig.playerData.get(uuid).playerName = name;
+                accConfig.save();
+            }
+        }
+        catch (SQLException | NoModuleException e)
+        {
+            e.printStackTrace();
+        }
+    }
+
+    public void updatePlayerJoinDate(final UUID uuid, final String date)
+    {
+        try
+        {
+            if (plugin.getModuleContainer().isModuleLoaded(DatabaseModule.ID))
+            {
+                updatePlayerAttribute(uuid, JOIN_DATE_COLUMN, date, true);
+            }
+            else
+            {
+
+                final AccountConfigData accConfig = getAccountConfig();
+                accConfig.playerData.get(uuid).joinDate = CommonUtils.dateFormat(date);
+                accConfig.save();
+            }
+        }
+        catch (SQLException | NoModuleException e)
+        {
+            e.printStackTrace();
+        }
+    }
+
+    public void updatePlayerLastDate(final UUID uuid, final String date)
+    {
+        try
+        {
+            if (plugin.getModuleContainer().isModuleLoaded(DatabaseModule.ID))
+            {
+                updatePlayerAttribute(uuid, LAST_JOIN_COLUMN, date, true);
+            }
+            else
+            {
+                final AccountConfigData accConfig = getAccountConfig();
+                accConfig.playerData.get(uuid).lastVisit = CommonUtils.dateFormat(date);
+                accConfig.save();
+            }
+        }
+        catch (SQLException | NoModuleException e)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1166)